File: field/src/test/java/field/BasicFieldTest.java
package field;

import org.junit.jupiter.api.Test;
import type.StringType;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;

class BasicFieldTest {

    @Test
    void createSqlPart() {
        BasicField basicField = BasicField.builder()
                .name("field_name")
                .type(new StringType())
                .isLast(true)
                .build();

        String expected = "field_name string";
        String actual = basicField.createSqlPart();
        assertThat(actual, is(expected));
    }
}

File: kotlin-version/src/test/kotlin/json/schema/hive/BasicFieldTest.kt
package json.schema.hive

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.Is.`is`
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class BasicFieldTest {

    @Test
    @DisplayName("tests the creation of SQL string")
    fun createSqlPart() {
        val basicField = BasicField(name = "hahaha", type = StringType(), isLast = false)
        val actual = basicField.createSqlPart()
        assertThat(actual, `is`("hahaha string,"))
    }
}
